### PR TITLE
Create configuration and database routing

### PR DESCRIPTION
**Pull request by EliuX**
This PR:

- Upgrades `pytest` to version `5.2.0`, so we can use `pytest-mock`.
- Tries out a mock in time_enties_namespace_test.py
- Add support for configurations. By default it works with `time_tracker_api.config.DefaultConfig`.
- Creates a configuration for a `whatever` database that can be taken as an example of what the access to any database should look like in a design/architecture that can be extensible. 
- whatever_repository.py how any database implementation (MongoDB, SQL databases, etc.) can be easily hooked using a generic Repository pattern.

In addition, I added a refactoring of the API models for what we thought would be a better naming system.

### CODE REVIEW

File: tests/time_entries/time_enties_namespace_test.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from flask import json
+
+def test_list_should_return_empty_array(mocker, client):
```
Comment (Angeluz-07) [inline tests/time_entries/time_enties_namespace_test.py]: Could we have type hints here?
Also, check the name of the file, I think It should be 'time_entries_namespace_test' or why not just 'time_entries_test' ?

File: time_tracker_api/__init__.py [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
+               config_data=None):
     flask_app = Flask(__name__)
 
+    init_app_config(flask_app, config_path, config_data)
     init_app(flask_app)
 
     return flask_app
 
 
+def init_app_config(app, config_path, config_data=None):
```
Comment (Angeluz-07) [inline time_tracker_api/__init__.py]: Could we have type hints here.

File: time_tracker_api/database.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Put here your utils and class independent of
+the database solution
+"""
+from flask import Flask
+
+RepositoryModel = None
+
+
+def init_app(app: Flask) -> None:
```
Comment (Angeluz-07) [inline time_tracker_api/database.py]: Why not call this function 'init_database' directly ?

File: time_tracker_api/__init__.py (hunk 2 above)
Comment (EliuX) [reply time_tracker_api/__init__.py]: Sure

File: time_tracker_api/database.py (hunk 3 above)
Comment (EliuX) [reply time_tracker_api/database.py]: There is some sort of standard that some objects like the APIs in flask Restplus have a function called init_app so they take the Flask app as a parameter and they do whatever initialization logic they need to be ready to be used by that app. So whatever module or package you add if you want to make it like pluggable to Flask this is pretty much the way to go

File: tests/time_entries/time_enties_namespace_test.py (hunk 1 above)
Comment (EliuX) [reply tests/time_entries/time_enties_namespace_test.py]: Ok

**Pull request merged**